src/Sentry/Protocol/SentryEvent.cs

```diff
@@ -117,6 +121,72 @@ public class SentryEvent : BaseScope
         /// </summary>
         public IDictionary<string, string> Modules => InternalModules ??= new Dictionary<string, string>();
 
+        /// <inheritdoc />
+        [DataMember(Name = "level", EmitDefaultValue = false)]
+        public SentryLevel? Level { get; set; }
+
+        /// <inheritdoc />
+        [DataMember(Name = "transaction", EmitDefaultValue = false)]
+        public string? Transaction { get; set; }
+
+        [DataMember(Name = "request", EmitDefaultValue = false)]
+        private Request? _request;
+
+        /// <inheritdoc />
+        public Request Request
+        {
+            get => _request ??= new Request();
+            set => _request = value;
+        }
+
+        [DataMember(Name = "contexts", EmitDefaultValue = false)]
+        private Contexts? _contexts;
+
+        /// <inheritdoc />
+        public Contexts Contexts
+        {
+            get => _contexts ??= new Contexts();
+            set => _contexts = value;
+        }
+
+        [DataMember(Name = "user", EmitDefaultValue = false)]
+        private User? _user;
+
+        /// <inheritdoc />
+        public User User
+        {
+            get => _user ??= new User();
+            set => _user = value;
+        }
+
+        /// <inheritdoc />
+        [DataMember(Name = "environment", EmitDefaultValue = false)]
+        public string? Environment { get; set; }
+
+        /// <inheritdoc />
+        [DataMember(Name = "sdk", EmitDefaultValue = false)]
+        public SdkVersion? Sdk { get; internal set; } = new SdkVersion();
```

Comment:
Actually this shouldn't be nullable:

```suggestion
        public SdkVersion Sdk { get; internal set; } = new SdkVersion();
```